src/VirtoCommerce.CustomerExportImportModule.Data/ExportImport/GenericClassMap.cs

```diff
@@ -0,0 +1,70 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using CsvHelper.Configuration;
+using VirtoCommerce.Platform.Core.Common;
+using VirtoCommerce.Platform.Core.DynamicProperties;
+
+namespace VirtoCommerce.CustomerExportImportModule.Data.ExportImport
+{
+    public sealed class GenericClassMap<T> : ClassMap<T>
+    {
+        public GenericClassMap(string[] dynamicProperties = null)
+        {
+            AutoMap(new Configuration() { Delimiter = ";", CultureInfo = CultureInfo.InvariantCulture });
+
+            var exportedType = ClassType;
+            var columnIndex = MemberMaps.Count - 1;
+
+            var typeHasDynamicProperties = exportedType.GetInterfaces().Contains(typeof(IHasDynamicProperties));
+
+            if (!dynamicProperties.IsNullOrEmpty() && typeHasDynamicProperties)
+            {
+                var currentClassMap = this;
+
+                // Exporting multiple csv fields from the same property (which is a collection)
+                foreach (var dynamicProperty in dynamicProperties)
+                {
+                    // create CsvPropertyMap manually, because this.Map(x =>...) does not allow
+                    // to export multiple entries for the same property
+                    var propertyValuesInfo = exportedType.GetProperty(nameof(IHasDynamicProperties.DynamicProperties));
+                    var csvPropertyMap = MemberMap.CreateGeneric(exportedType, propertyValuesInfo);
+                    csvPropertyMap.Name(dynamicProperty);
+
+                    csvPropertyMap.Data.Index = ++columnIndex;
```

Comment:
I'll prefer to write `var columnIndex = MemberMaps.Count` and `csvPropertyMap.DataIndex = columnIndex++`